# Finance Job Market Analysis

This notebook is where I will be cleaning the data I scraped from LinkedIn using [linkedin-job-scraper](https://github.com/Daneski13/linkedin-job-scraper).

## Data Cleaning

### Import Data

In [9]:
import pandas as pd
import glob

files = glob.glob("data/*.csv")
                  
dfs = []
for file in files:
    dfs.append(pd.read_csv(file))

linkedin = pd.concat(dfs)    
display(linkedin.head())
linkedin.describe()

date_scraped                                              title  \
0   2022-10-04                     Quantitative analyst (finance)   
1   2022-10-03                  Accounting and finance associates   
2   2022-09-28  Associate/ Consulting Associate - Litigation, ...   
3   2022-09-28  Associate/ Consulting Associate - Litigation, ...   
4   2022-10-04                  Senior Financial Analyst (Remote)   

                                            full_url  \
0  https://www.linkedin.com/jobs/view/quantitativ...   
1  https://www.linkedin.com/jobs/view/accounting-...   
2  https://www.linkedin.com/jobs/view/associate-c...   
3  https://www.linkedin.com/jobs/view/associate-c...   
4  https://www.linkedin.com/jobs/view/senior-fina...   

                             company  \
0  Lucas Group, A Korn Ferry Company   
1          EVERESTX Talent Solutions   
2           Charles River Associates   
3           Charles River Associates   
4               Capital Search Group   

                                         company_url  \
0  https://www.linkedin.com/company/lucas-group?t...   
1  https://www.linkedin.com/company/everestxtalen...   
2  https://www.linkedin.com/company/charles-river...   
3  https://www.linkedin.com/company/charles-river...   
4  https://www.linkedin.com/company/capital-searc...   

                      location  \
0                Charlotte, NC   
1  Pennsylvania, United States   
2               Washington, DC   
3                  Chicago, IL   
4                   McLean, VA   

                                         description seniority_level  \
0  \n        <p>Lucas Group, a Korn Ferry company...       Associate   
1  \n        <strong>Overview of the Role:</stron...    Not Assigned   
2  \n        <strong>About Charles River Associat...     Entry level   
3  \n        <strong>About Charles River Associat...     Entry level   
4  \n        Microsoft has become a corporate lea...    Not Assigned   

  employment_type  job_function                        industries  
0       Full-time       Finance                           Banking  
1    Not Assigned  Not Assigned                      Not Assigned  
2       Full-time         Legal  Business Consulting and Services  
3       Full-time         Legal  Business Consulting and Services  
4    Not Assigned  Not Assigned                      Not Assigned

date_scraped                    title  \
count         60017                    60017   
unique            7                    12294   
top      2022-09-30  Remote Tax Professional   
freq          11299                     2443   

                                                 full_url       company  \
count                                               60017         60016   
unique                                              29607          7205   
top     https://www.linkedin.com/jobs/view/finance-and...  Aston Carter   
freq                                                    7          3339   

                                              company_url       location  \
count                                               60017          60017   
unique                                               7213           3071   
top     https://www.linkedin.com/company/aston-carter?...  United States   
freq                                                 3339           3738   

                                              description seniority_level  \
count                                               60016           51890   
unique                                              14787               6   
top     \n        <strong><u>What You'll Do...<br><br>...       Associate   
freq                                                 2474           23092   

       employment_type                     job_function    industries  
count            52288                            52288         52288  
unique               4                              693          1009  
top          Full-time  Accounting/Auditing and Finance  Not Assigned  
freq             40980                            10160          9708

### Drop Duplicates

Out of the 31,383 job listings that were scraped, 29,607 were unique

In [10]:
linkedin.drop_duplicates(subset="full_url", inplace=True)
linkedin.describe()

date_scraped                    title  \
count         29607                    29607   
unique            7                    12294   
top      2022-09-30  Remote Tax Professional   
freq           5669                     1214   

                                                 full_url       company  \
count                                               29607         29607   
unique                                              29607          7205   
top     https://www.linkedin.com/jobs/view/quantitativ...  Aston Carter   
freq                                                    1          1749   

                                              company_url       location  \
count                                               29607          29607   
unique                                               7213           3071   
top     https://www.linkedin.com/company/aston-carter?...  United States   
freq                                                 1749           1833   

                                              description seniority_level  \
count                                               29606           29400   
unique                                              14647               6   
top     \n        <strong><u>What You'll Do...<br><br>...       Associate   
freq                                                 1229           11325   

       employment_type  job_function    industries  
count            29422         29422         29422  
unique               4           691          1000  
top          Full-time  Not Assigned  Not Assigned  
freq             20245          8380          8381

### Handle Nulls

In [11]:
linkedin.isna().sum()

date_scraped         0
title                0
full_url             0
company              0
company_url          0
location             0
description          1
seniority_level    207
employment_type    185
job_function       185
industries         185
dtype: int64

First lets drop the row with the missing description.

In [12]:
linkedin.dropna(subset="description", inplace=True)
linkedin.isna().sum()

date_scraped         0
title                0
full_url             0
company              0
company_url          0
location             0
description          0
seniority_level    207
employment_type    185
job_function       185
industries         185
dtype: int64

I will fill the rest of the missing data with the string "Not Assigned"

In [13]:
linkedin.fillna("Not Assigned", inplace=True)

### Data Types

In [14]:
linkedin.info(show_counts=False)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 29606 entries, 0 to 4625
Data columns (total 11 columns):
 #   Column           Dtype 
---  ------           ----- 
 0   date_scraped     object
 1   title            object
 2   full_url         object
 3   company          object
 4   company_url      object
 5   location         object
 6   description      object
 7   seniority_level  object
 8   employment_type  object
 9   job_function     object
 10  industries       object
dtypes: object(11)
memory usage: 2.7+ MB


### Remove Unnecessary listings

If a listing does not mention finance in it's title or it's description, the listing can be discarded as only finance related job postings are relevant.

In [15]:
# Drop rows that do not mention finance, "financ" is chosen to include words such as "financial"
def drop(x):
    if 'financ' in x["description"].lower() or 'financ' in x["title"].lower():
        return True
    return False

linkedin = linkedin[linkedin.apply(drop, axis=1)]

### Export cleaned data

After cleaning we are left with 28,634 listings out of the 31,383 originally scraped


In [16]:
display(linkedin.shape[0])
linkedin.to_csv("data/cleaned.csv", index=False)

28634